lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetSchemaConverter.java

```diff
@@ -104,22 +105,19 @@ private org.apache.parquet.schema.Type getPrimitiveType(Type type, String name,
             DecimalType decimalType = (DecimalType) type;
             if (decimalType.getPrecision() <= 9) {
                 return Types.optional(PrimitiveType.PrimitiveTypeName.INT32)
-                        .as(OriginalType.DECIMAL)
-                        .precision(decimalType.getPrecision())
-                        .scale(decimalType.getScale()).named(name);
+                        .as(decimalType(decimalType.getScale(), decimalType.getPrecision()))
+                        .named(name);
             }
             else if (decimalType.isShort()) {
                 return Types.optional(PrimitiveType.PrimitiveTypeName.INT64)
-                        .as(OriginalType.DECIMAL)
-                        .precision(decimalType.getPrecision())
-                        .scale(decimalType.getScale()).named(name);
+                        .as(decimalType(decimalType.getScale(), decimalType.getPrecision()))
+                        .named(name);
             }
             else {
                 return Types.optional(PrimitiveType.PrimitiveTypeName.FIXED_LEN_BYTE_ARRAY)
                         .length(16)
-                        .as(OriginalType.DECIMAL)
-                        .precision(decimalType.getPrecision())
-                        .scale(decimalType.getScale()).named(name);
+                        .as(decimalType(decimalType.getScale(), decimalType.getPrecision()))
```

Comment:
Does this result in same metadata in the file footer?
ie are both original type and logical annotation present?

Reply:
I checked that the schema in the footer is the same because `PrimitiveType#getOriginalType` will just give back OriginalType.DECIMAL from `DecimalLogicalTypeAnnotation#toOriginalType`.
In general, LogicalTypes are supposed to be backward compatible with OriginalType.
Hive has also switched to using these https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/io/parquet/convert/HiveSchemaConverter.java#L127